pkg/storage/secret/metadata/keeper_store.go

```diff
@@ -71,29 +89,71 @@ func (s *keeperMetadataStorage) Read(ctx context.Context, namespace xkube.Namesp
 		return nil, fmt.Errorf("missing auth info in context")
 	}
 
-	row := &keeperDB{Name: name, Namespace: namespace.String()}
-	err := s.db.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		found, err := sess.Get(row)
+	k, err := read(ctx, namespace.String(), name, s)
+	if err != nil {
+		return nil, err
+	}
+
+	var keeper *secretv0alpha1.Keeper
+
+	keeper, err = k.toKubernetes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to kubernetes object: %w", err)
+	}
+
+	if keeper != nil {
+		return keeper, nil
+	}
+
+	return nil, contracts.ErrKeeperNotFound
+}
+
+func read(ctx context.Context, namespace string, name string, s *keeperMetadataStorage) (*keeperDB, error) {
+	req := &readKeeper{
+		SQLTemplate: sqltemplate.New(s.dialect),
+		Namespace:   namespace,
+		Name:        name,
+	}
+	q, err := sqltemplate.Execute(sqlKeeperRead, req)
+	if err != nil {
+		return nil, fmt.Errorf("read template %q: %w", q, err)
```

Comment:
I don't think we should log the template string output, well here anyway this will be empty, if you check the function definition, if there's an error, the output is ""

Reply:
you are right about that, I took this error from other examples of other parts of the code, but it wont print anything useful. I will change it to just print the name of the template to have more information

Reply:
fixed error messages in https://github.com/grafana/grafana/pull/102530/commits/eefaafd010a932ccd08e57bcc1528c073973fdfb